lake/data/seekindex.go

```diff
@@ -9,48 +9,43 @@ import (
 	"github.com/brimdata/zed/order"
 	"github.com/brimdata/zed/pkg/storage"
 	"github.com/brimdata/zed/runtime/expr"
-	"github.com/brimdata/zed/runtime/expr/extent"
 	"github.com/brimdata/zed/zio/zngio"
 	"github.com/brimdata/zed/zson"
 )
 
 func LookupSeekRange(ctx context.Context, engine storage.Engine, path *storage.URI,
-	obj *Object, cmp expr.CompareFn, filter *expr.SpanFilter, countSpan extent.Span, o order.Which) (*seekindex.Range, error) {
+	obj *Object, pruner expr.Evaluator, o order.Which) ([]seekindex.Range, error) {
+	if pruner == nil {
+		// scan whole object
+		return nil, nil
+	}
 	r, err := engine.Get(ctx, obj.SeekIndexURI(path))
 	if err != nil {
 		return nil, err
 	}
 	defer r.Close()
+	var ranges []seekindex.Range
 	var rg *seekindex.Range
 	unmarshaler := zson.NewZNGUnmarshaler()
 	reader := zngio.NewReader(zed.NewContext(), r)
```

Comment:
We should close this ZNG reader.
```suggestion
	reader := zngio.NewReader(zed.NewContext(), r)
	defer reader.Close()
```